app/src/main/java/com/github/onedirection/Event.java

```diff
@@ -0,0 +1,103 @@
+package com.github.onedirection;
+
+class Event {
+
+    final private int id;
+    final private String name;
+    final private String location;
+    final private String date;
+    final private String start_time;
+    final private String end_time;
+
+
+    public Event(int id,String name, String location, String date, String start_time, String end_time){
+        if(name == null || location == null || date == null || start_time == null || end_time == null){
+            throw new IllegalArgumentException();
+        }
+        this.id = id;
+        this.name = name;
+        this.location = location;
+        this.date = date;
+        this.start_time = start_time;
+        this.end_time = end_time;
+    }
+
+    public Event set_name(String new_value){
+        if(new_value ==  null){
+            throw new IllegalArgumentException();
+        }
+        if(new_value == this.name){
+            return this;
+        }
+        return new Event(id,new_value,location,date,start_time,end_time);

```

Comment:
Shorter
```suggestion
        Objects.requireNonNull(new_value);
        return new_value == this.name ?
            this:
            new Event(id,new_value,location,date,start_time,end_time);

```
I would at least change the Null check (so that CodeClimate stops complaining).
The same applies to all setters.